pkg/utils/gardener/namespacedcloudprofile.go

```diff
@@ -0,0 +1,66 @@
+// SPDX-FileCopyrightText: 2024 SAP SE or an SAP affiliate company and Gardener contributors
+//
+// SPDX-License-Identifier: Apache-2.0
+
+package gardener
+
+import (
+	"context"
+	"fmt"
+
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	pkgclient "sigs.k8s.io/controller-runtime/pkg/client"
+
+	"github.com/gardener/gardener/pkg/api"
+	"github.com/gardener/gardener/pkg/apis/core"
+	gardencorev1beta1 "github.com/gardener/gardener/pkg/apis/core/v1beta1"
+	"github.com/gardener/gardener/pkg/apis/core/v1beta1/constants"
+)
+
+// GetCloudProfile gets determine whether a given CloudProfile name is a NamespacedCloudProfile or a CloudProfile and returns the appropriate object
+func GetCloudProfile(ctx context.Context, client pkgclient.Client, cloudProfileReference *gardencorev1beta1.CloudProfileReference, namespace string) (*gardencorev1beta1.CloudProfile, error) {
```

Comment:
```suggestion
func GetCloudProfile(ctx context.Context, reader client.Reader, shoot *gardencorev1beta1.Shoot) (*gardencorev1beta1.CloudProfile, error) {
```

I think we can improve the signature of this function:
- How do users know that they ought call `BuildCloudProfileReferenceV1Beta1` to pass the `cloudProfileReference`?
- Signature is simpler by just passing the shoot.